pkg/landscaper/controllers/installations/reconcile.go

```diff
@@ -83,6 +83,10 @@ func (c *Controller) reconcile(ctx context.Context, inst *lsv1alpha1.Installatio
 		return nil
 	}
 
+	err = c.ConstructImports(ctx, instOp)
```

Comment:
Hm, I think this is problematic. The import construction also verifies that all imports are up-to-date - and all installations this one depends on are in a final state - and fails if this is not the case. Inserting this check at this point could lead to the following scenario:
1. Installation A "runs" successfully (=> all deployitems/subinstallations end in 'Succeeded' state)
2. While A is running, B gets an update and starts processing. A depends on B.
3. After all of A's nested items have run, A tries to construct its exports. The check mentioned above kicks in and fails, because B is currently running and the value which A imports from B cannot be fetched.

=> Installation A has run successfully, but fails when creating the exports because B is running, which should not happen.

Reply:
I don't like the way this checks and execution are interwoven either and thought about refactoring this. 
But there might be a more general problem here:
Import construction pulls the import values from the cluster. If this happens twice - once at the beginning of the processing of an installation and once after it has finished, there is no way to be sure that we get the same import values both times, they could always change in between ...

Reply:
OK, my bad, it is already the case that the exports are not generated if the imports have changed, so this change doesn't actually break something.